pytorch_lightning/strategies/launchers/subprocess_script_hydra.py

```diff
@@ -0,0 +1,82 @@
+import os
+from typing import List, Optional, Tuple
+
+import __main__
+import torch
+
+from pytorch_lightning.strategies.launchers.subprocess_script import _SubprocessScriptLauncher
+from pytorch_lightning.utilities import _HYDRA_AVAILABLE
+
+if _HYDRA_AVAILABLE:
+    from hydra.core.hydra_config import HydraConfig
+    from hydra.utils import to_absolute_path
+
+
+class _SubprocessScriptHydraLauncher(_SubprocessScriptLauncher):
```

Comment:
I would put this at the beginning

```suggestion
class _HydraSubprocessScriptLauncher(_SubprocessScriptLauncher):
```

Or even

```suggestion
class _HydraLauncher(_SubprocessScriptLauncher):
```

Reply:
`_HydraSubprocessScriptLauncher` looks better.

Reply:
I like the idea of a hydra specific launcher. 
However, my suggestion is not to subclass the regular process launcher. We are moving away from that pattern now in the strategies too (#11863). 

I suggest to put both launchers into the same file and simply share code through logical pieces of utility function in that same file.